WordPress/gutenberg
# [RNMobile] - E2E Simplify heading and lists blocks functions

## What?
- This PR is a follow-up to https://github.com/WordPress/gutenberg/pull/40404, this PR will attempt to simplify other text block implementations. I earlier missed that it was implemented separately for paragraph/heading/list and did not want to introduce more changes to the previous PR.
- With the changes, we can also now re-enable the slash inserter tests and close https://github.com/wordpress-mobile/gutenberg-mobile/issues/3699
- Also included in the PR is rearranging the location of some functions in the `editor-page.js` so that it would be easier to find

## Why?
To reduce E2E test flakiness when running in the CI. Also to simplify the implementation, we can now use 1 function instead of many for text-based blocks. Also will be able to remove the extra click that was previously needed on Android on the test level, as it has since been covered on the function itself.

## How?
For Paragraph and Heading blocks, we can now use `getTextBlockAtPosition()` to get block position. Would still have to separate the List block because of locator differences. To access List block, use `getListBlock()`. To type text into blocks, all text blocks can now use `typeTextToTextBlock()`

## Testing Instructions
Test impacted in this PR should now work both locally and in the CI on both platforms (change platform from `android` to `ios` to test iOS)
```
TEST_RN_PLATFORM=android npm run native device-tests gutenberg-editor-heading-@canary.test
TEST_RN_PLATFORM=android npm run native device-tests gutenberg-editor-lists-@canary.test
TEST_RN_PLATFORM=android npm run native device-tests gutenberg-editor-lists-end.test.js
TEST_RN_PLATFORM=android npm run native device-tests gutenberg-editor-lists.test.js
TEST_RN_PLATFORM=android npm run native device-tests gutenberg-editor-paragraph.test
TEST_RN_PLATFORM=android npm run native device-tests gutenberg-editor-paste.test
TEST_RN_PLATFORM=android npm run native device-tests gutenberg-editor-rotation.test
TEST_RN_PLATFORM=android npm run native device-tests gutenberg-editor-slash-inserter-@canary.test
```
## Screenshots or screencast
